src/ui.rs

```diff
@@ -115,40 +115,53 @@ impl ThemedWidget for &Test {
                 }))
                 // current word
                 .chain({
-                    let progress_ind = self.words[self.current_word]
-                        .progress
-                        .len()
-                        .min(self.words[self.current_word].text.len());
-
-                    let correct = self.words[self.current_word]
-                        .text
-                        .starts_with(&self.words[self.current_word].progress[..]);
-
-                    let (typed, untyped) =
-                        self.words[self.current_word]
-                            .text
-                            .split_at(ceil_char_boundary(
```

Comment:
This results in cargo throwing the warning:
```
warning: function `ceil_char_boundary` is never used
   --> src/ui.rs:346:4
    |
346 | fn ceil_char_boundary(string: &str, index: usize) -> usize {
    |    ^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default
```

Reply:
I removed the unused code.